src/main/java/uk/gov/hmcts/reform/et/syaapi/service/CaseDocumentService.java

```diff
@@ -0,0 +1,211 @@
+            HttpMethod.POST,
+            request,
+            DocumentUploadResponse.class
+        );
+    }
+
+    private HttpHeaders getHttpHeaders(String authToken) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HttpHeaders.AUTHORIZATION, authToken);
+        headers.add(SERVICE_AUTHORIZATION, authTokenGenerator.generate());
+        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+        return headers;
+    }
+
+    private void validateFile(MultipartFile file) throws CaseDocumentException, IOException {
+        String filename = file.getOriginalFilename();
+        if (filename == null || filename.isEmpty()) {
+            throw new CaseDocumentException(VALIDATE_FILE_EXCEPTION_MESSAGE);
+        }
+
+        Matcher matcher = FILE_NAME_PATTERN.matcher(filename);
+        if (!matcher.matches()) {
+            throw new CaseDocumentException(VALIDATE_FILE_EXCEPTION_MESSAGE);
+        }
+
+        Tika tika = new Tika();
+        String detectedType = tika.detect(file.getBytes());
+
+        if (!detectedType.equals(file.getContentType())) {
+            throw new CaseDocumentException(VALIDATE_FILE_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private CaseDocument validateResponse(DocumentUploadResponse response, String originalFilename)
+        throws CaseDocumentException {
+        if (response.documents == null || response.isEmpty()) {
+            throw new CaseDocumentException(UPLOAD_FILE_EXCEPTION_MESSAGE + originalFilename);
+        }
+
+        CaseDocument document = response.getDocuments().stream()
+            .findFirst()
+            .orElseThrow(() -> new CaseDocumentException(UPLOAD_FILE_EXCEPTION_MESSAGE + originalFilename));
+
+        String uri = getUriFromFile(document).toString();
+
+        Matcher matcher = HTTPS_URL_PATTERN.matcher(uri);
+        if (!matcher.matches()) {
+            throw new CaseDocumentException(UPLOAD_FILE_EXCEPTION_MESSAGE + originalFilename);
+        }
+
+        return document;
+    }
+
+    private MultiValueMap<String, Object> generateUploadRequest(String caseTypeId,
+                                                                MultipartFile file) throws IOException {
+        ByteArrayResource fileAsResource = new ByteArrayResource(file.getBytes());
+
+        MultiValueMap<String, Object> body = new LinkedMultiValueMap<>();
+        body.add("files", fileAsResource);
+        body.add("classification", PUBLIC.toString());
+        body.add("caseTypeId", caseTypeId);
+        body.add("jurisdictionId", JURISDICTION);
+
+        return body;
+    }
+
+    @Data
+    private static class DocumentUploadResponse {
+        private List<CaseDocument> documents;
+
+        /**
+         * Accessor for revealing if the response has any documents.
+         *
+         * @return a boolean that is false if documents list is empty
+         */
+        public Boolean isEmpty() {
+            return documents.isEmpty();
+        }
```

Comment:
Surely not necessary to have this method too?
Maybe calling getDocuments().isEmpty() would do the same thing?  Or is this PMD moaning again?

Reply:
This was just addressing a PMD error

Reply:
I'm keen to see the PMD error, let's look at this one together ;)